modules/cluster/src/main/java/io/fluo/cluster/util/FluoInstall.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015 Fluo authors (see AUTHORS)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.fluo.cluster.util;
+
+import java.io.File;
+
+import io.fluo.api.config.FluoConfiguration;
+import io.fluo.api.exceptions.FluoException;
+
+public class FluoInstall {
+
+  private String fluoHomeDir;
+
+  public FluoInstall(String fluoHomeDir) {
+    this.fluoHomeDir = fluoHomeDir;
+  }
+
+  public String getFluoConfDir() {
+    return fluoHomeDir + "/conf";
+  }
+
+  public String getFluoPropsPath() {
+    return getFluoConfDir() + "/fluo.properties";
+  }
+
+  public String getLibDir() {
+    return fluoHomeDir + "/lib";
+  }
+
+  public String getAppsDir() {
+    return fluoHomeDir + "/apps";
+  }
+
+  public String getAppConfDir(String appName) {
+    return String.format("%s/%s/conf", getAppsDir(), appName);
+  }
+
+  public String getAppLibDir(String appName) {
+    return String.format("%s/%s/lib", getAppsDir(), appName);
+  }
+
+  public String getAppPropsPath(String appName) {
+    return getAppConfDir(appName) + "/fluo.properties";
+  }
+
+  public void verifyFluoInstall() {
+    verifyInstallPath(fluoHomeDir);
+    verifyInstallPath(getLibDir());
+    verifyInstallPath(getFluoConfDir());
+    verifyInstallPath(getFluoPropsPath());
+  }
+
+  public void verifyAppInstall(String appName) {
+    verifyFluoInstall();
+    verifyAppPath(appName, getAppsDir());
+    verifyAppPath(appName, getAppConfDir(appName));
+    verifyAppPath(appName, getAppPropsPath(appName));
+  }
+
+  public FluoConfiguration getAppConfiguration(String appName) {
+    verifyAppInstall(appName);
+    FluoConfiguration config = new FluoConfiguration(new File(getAppPropsPath(appName)));
+    if (!config.getApplicationName().equals(appName)) {
+      throw new FluoException("Application name in config '" + config.getApplicationName()
+          + "' does not match given appName: " + appName);
+    }
+    return config;
+  }
+
+  public FluoConfiguration getFluoConfiguration() {
+    verifyFluoInstall();
+    return new FluoConfiguration(new File(getFluoPropsPath()));
+  }
+
+  public FluoConfiguration resolveFluoConfiguration(String appName) {
```

Comment:
Not sure if this would be the best place to print it.  I was thinking it may be useful to print which config file it ended up using.